test/test_utils.hpp

```diff
@@ -25,6 +25,22 @@
 #ifndef XSIMD_TEST_UTILS_HPP
 #define XSIMD_TEST_UTILS_HPP
 
+/**************************
+ * AppleClang workarounds *
+ *************************/
+
+// AppleClang is known for having precision issues
+// between AVX and AVX2. This also seems to happen
+// in M1.
+struct precision_t
+{
+#ifdef __apple_build_version__
```

Comment:
Should I add an architecture check here? e.g. `&& (defined(__AVX2__) || defined(NEON64))` (or the xsimd macros thereof)

Reply:
I'd say that the more refined the check is, the better. So yeah, please go ahead